src/Grid.php

```diff
@@ -513,6 +513,37 @@ public function addModalAction($button, $title, \Closure $callback, $args = [])
         return $this->getActionButtons()->addModal($button, $title, $callback, $this, $args);
     }
 
+    /**
+     * Similar to addModalAction but apply to a multiple recors selection and display in menu.
+     * When menu item is clicked, modal is displayed with the $title and $callback is executed through VirtualPage.
+     *
+     * @param string|array|MenuItem $item
+     * @param \Closure              $callback function (View $page) {...
+     * @param array                 $args     extra url argument for callback
+     *
+     * @return Jquery
+     */
+    public function addModalBulkAction($item, \Closure $callback, $args = [])
+    {
+        if (!$this->menu) {
+            throw new Exception('Unable to add Modal Bulk Action without Menu');
+        }
+
+        $owner = $this->getOwner();
+
+        if (is_string($item)) {
+            $item = ['title' => $item];
+        }
+
+        $modal = Modal::addTo($owner, [$item[0]]);
```

Comment:
no need to assing to `$owner` first, use `$this->getOwner()` directly

Reply:
ok, done